### PR TITLE
Add check if LocalAccountTokenFilterPolicy is enabled

### LINKED ISSUE #4033: Enable-PSRemoting doesn't enable LocalAccountTokenFilterPolicy if WinRM Firewall rule is enabled
https://github.com/PowerShell/PowerShell/blob/7a55bf98b2370ab4817ab2533cb67673053ee446/src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs#L4045

Checks if WinRMQuickConfig (winrm qc) is needed to be run to enable PowerShell remoting.  It checks if WinRM service is running, startup mode is automatic, listener created, and if the WinRM firewall rule is enabled, but doesn't check if LocalAccountTokenFilterPolicy is enabled.  This causes an issue on WS2016 where WinRM is enabled by default so Enable-PSRemoting doesn't set the LocalAccountTokenFilterPolicy registry setting compared to WS2012R2.

Reported by customer

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Fix #4033.

Add the check (LocalAccountTokenFilterPolicy  == 1 in registry) in internal Test-WinRMQuickConfigNeeded function.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by iSazonov** [comment]
Should it be back ported to 6.2?

**Comment by iSazonov** [comment]
@SteveL-MSFT @PaulHigin Could you please look the small fix?

File: src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs [hunk 1]
```diff
@@ -4056,6 +4056,11 @@ public sealed class EnablePSSessionConfigurationCommand : PSCmdlet
         $winrmQuickConfigNeeded = $true
     }}
 
+    # check if LocalAccountTokenFilterPolicy is enabled
```
[the remark this answers is not in the document]
Comment (TravisEz13) [reply src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs]: @iSazonov 	 I think one solution to this problem is to prompt the user if they really want to do this by putting this behind a ShouldProcess High Impact.

[the remark this answers is not in the document]
Comment (TravisEz13) [reply src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs]: Also, running I ran `Set-WSManQuickConfig -Force -SkipNetworkProfileCheck` and it did **not** set this policy for me.

[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs]: I re-read the original issue and also https://support.microsoft.com/en-us/help/951016/description-of-user-account-control-and-remote-restrictions-in-windows
and my understanding now is that Win2016 is more secure than Win2012R2. _The option only affects local accounts, not domain accounts,_ and doesn't break remote management scenarios.
So we must reject the fix at all.

/cc @SteveL-MSFT

[the remark this answers is not in the document]
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs]: I'm ok with reject the change and closing the issue.  Thanks.

**Pull request closed without merging**